via/static/scripts/video_player/hooks/use-side-by-side-layout.ts

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useEffect, useState } from 'preact/hooks';
+
+type SidebarLayout = {
+  /** Whether sidebar is open or closed */
+  expanded: boolean;
+  /** Current width of sidebar in pixels */
+  width: number;
+  /** Current height of sidebar in pixels */
+  height: number;
+  /** Width of controls (toolbar, bucket bar) on the edge of the sidebar */
+  toolbarWidth: number;
+};
+
+type LayoutChangeEventDetail = {
+  sideBySideActive: boolean;
+  sidebarLayout: SidebarLayout;
+};
+
+type LayoutChangeEvent = CustomEvent & { detail: LayoutChangeEventDetail };
+
+// Minimum body width at which laying out side-by-side should be considered
+export const SIDEBYSIDE_THRESHOLD = 875;
+// The sidebar reports its toolbar width as 33px. This is a custom padding
+// amount that allows this app to nestle up more cloesly to the sidebar when
+// closed.
+export const TOOLBAR_WIDTH = 22;
+
+/**
+ * Adjust the available space to the right of app content to make room for
+ * the sidebar when plausible. This mimics the client's "side-by-side" mode,
+ * tuned specifically for this application.
+ *
+ * Side-by-side layout will be re-computed whenever sidebar state via the
+ * `hypothesis:layoutchange` event changes meaningfully, or when the window
+ * is resized.
+ */
+export function useSideBySideLayout() {
+  const [sideBySideApplicable, setSideBySideApplicable] = useState(false);
+  const [sufficientSpace, setSufficientSpace] = useState(
+    document.body.clientWidth >= SIDEBYSIDE_THRESHOLD
+  );
+  const [sidebarWidth, setSidebarWidth] = useState(0);
```

Comment:
The layout event fires frequently, as could resize, but my aim with these is to have state that don't change value too often. This was the minimum set I could come up with (given that a resize handler would need some retained sidebar state to reference), but I'm open to suggestions.

Reply:
Maybe this could benefit from `useReducer`? It's a hook I think I have used zero times in production code, but dropping the suggestion here in case it makes sense 🤷🏼‍♂️

Reply:
I think this is fine as-is. I've tuned the state primitives to represent "meaningful state change": if any of these state values change, we legitimately do need to re-evaluate our side-by-side layout.